src/common/state/ColorControlPointList.C

```diff
@@ -709,23 +710,32 @@ ColorControlPointList::SetExternalFlag(bool externalFlag_)
 // Method: ColorControlPointList::SetTagNames
 //
 // Purpose:
-//   Setter for names.
+//   Setter for tagNames.
 //
-// Note:       There needs to be a custom setter to make sure that every time
-//             the tagNames are set, tagChangesMade is set to true.
+// Note:       We need a custom setter to vet the tag names before they are
+//             added to the tagNames vector.
 //
 // Programmer: Justin Privitera
 // Creation:   Wed Jun 29 16:38:18 PDT 2022
 //
 // Modifications:
+//     Justin Privitera, Thu Aug 25 15:04:55 PDT 2022
+//     Validate tags before adding them to the list.
 //
 // ****************************************************************************
 
 void
 ColorControlPointList::SetTagNames(const stringVector &tagNames_)
 {
-    tagNames = tagNames_;
-    tagChangesMade = true;
+    std::for_each(tagNames_.begin(), tagNames_.end(),
+        [this, tagNames_](std::string currtag)
+        {
+            auto result{ValidateTag(currtag)};
+            if (result.first)
+                AddTag(currtag);
+            else
+                debug1 << "ColorControlPointList WARNING: " << result.second;
```

Comment:
I took a look at stuff in `src/common/state` and the truth is, we really don't have a ton of precedent for error handling and/or messaging from state objects. I greped `-i` for `debug`, `except` and `avtcallback` and see a few instances of each in a few cases.

* `debugX` messages will almost certainly never be seen by any user, regardless of `X` (`1...5`). Debug messages should really be viewed as more for developer consumption anyways...not end-users. I dunno if your intent is for end-users to see these but if so, I don't think `debugX` is the way to go.
* `EXCEPTION(...)` is used in `DBOptionsAttributes`. Looks like Hank put those in...18 years ago. User's are a little more likely to see these I think because they sometimes result in information in the GUI's **Output** window or the CLI's `stderr`. But, exceptions also potentially cause VisIt to behave badly thereafter, depending on when they are thrown and how they are handled and since state objects are used everywhere...not sure this is best idea either.
* Looks like `avtCallback::IssueWarning` is already used in `ColorControlPointList` and in `ColorTableAttributes` but only for the `ProcessOldVersions` (of xml session/settings files). This is indeed for end-user messaging.

If your goal is user messaging, whada think about replacing messages to `debugX` here with returning error strings and/or status instead and let caller handle the messaging.

Otherwise, I think having `debugX` outputs is fine as is.

Reply:
Hmmm... I guess the question is whether or not we want these warnings to be seen by users. I agree that we definitely don't want an exception, since encountering a bad tag name should just result in that tag name being skipped.

This particular warning will only be triggered when visit reads the `.ct` files in the `resources` directory and encounters a bad tag in one of those, which is very unlikely. A user would have had to either edit those files by hand or put new files there. In that case it could be helpful to have a warning that is visible to users. I guess I'm on the fence. Thoughts?

Reply:
The caller is likely inside the same file (when this warning is triggered, that is), so I don't think letting someone else handle the warning message is an option.

Reply:
I lean towards not worrying about messaging user and having things silently not work when invalid chars are in play. So, I'd vote for eliminating lines to send anything to `debugX`.

Reply:
Ok, I'll leave as is then.